src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -103,12 +103,11 @@ public Dictionary(int capacity, IEqualityComparer<TKey>? comparer)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.collection);
             }
 
-            Copy(collection, this);
+            Copy(collection);
         }
 
-        private static void Copy(IEnumerable<KeyValuePair<TKey, TValue>> collection, Dictionary<TKey, TValue> dest)
+        private void Copy(IEnumerable<KeyValuePair<TKey, TValue>> collection)
```

Comment:
The usual name for methods like this is AddRange

Reply:
Would need to add some extra checks for the fast path for a true AddRange (`.Length >= .Count`, `_count == 0`)

Debug check and throw; or add the tests and make it ready to be a public api?

Reply:
ok, let's keep it as is.

Reply:
Oh, already added Debug checks + comment :)

Reply:
Both Copy and AddRange looks goods to me. I will leave it up to you what you like better.

Reply:
AddRange works for me; also leaves door open to be extended to a public api